### ISSUE TITLE
[JENKINS-66161] Remove use of google.com and also make use of HTTPS url's or endpoint

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#12114]
<p>Hey Team / <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke" class="user-hover" rel="kohsuke">kohsuke</a>,</p>

<p>I would like to propose enhancement or improvement on two underrated issues in Jenkins that I noticed while deploying in enterprise systems.</p>
<ol>
	<li><b>Use of HTTP endpoints / URL and remove redirects</b> - SecOps are concerned to allowlist HTTP URLs and redirects today in some enterprise, and it is impossible to scan the huge volume in short period of time if the implementation of Jenkins is critical or time sensitive. We need to change this so that it will be seamless to deploy without losing the market for other cloud vendors over time, since many senior management would go for managed servers than approving ( even temporarily for install or upgrade. Redirects are waste of time, increases response time.</li>
	<li><b>Use of google.com URLs or use of sites that are not under *.jenkins.io for plugin should be changed to something reliable or trustworthy</b> - This is another red flag raised by SecOps and we need to remove use of google.com starting from checkConnectionURL() to anyplace it is not required. If it is really needed we may need to document it with what are the domain (full address) has to request to apply in allowed list on companies firewall on the cloud (applies to on prem too).</li>
</ol>

<p><b>Advantages</b></p>
<ol>
	<li>Increase in upgrade - Most enterprise may be running on old version of Jenkins without upgrading because these  firewall restrictions on HTTP or google.com will cause implementation failure and hard to rollback.</li>
	<li>Builds more trust in Enterprise space - It will help Jenkins to be consumed by many enterprise and management would not think of other cloud options, due to this security issues.</li>
	<li>More web traffic to Jenkins.io - More hits to Jenkins.io website if we shift from using google.com and indirectly it may help SEO and many other digital marketing advantages.</li>
</ol>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=prnam">prnam</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161" target="_blank">Remove use of google.com and also make use of HTTPS url's or endpoint</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Hey Team / <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke" class="user-hover" rel="kohsuke">kohsuke</a>,</p>

<p>I would like to propose enhancement or improvement on two underrated issues in Jenkins that I noticed while deploying in enterprise systems.</p>
<ol>
	<li><b>Use of HTTP endpoints / URL and remove redirects</b> - SecOps are concerned to allowlist HTTP URLs and redirects today in some enterprise, and it is impossible to scan the huge volume in short period of time if the implementation of Jenkins is critical or time sensitive. We need to change this so that it will be seamless to deploy without losing the market for other cloud vendors over time, since many senior management would go for managed servers than approving ( even temporarily for install or upgrade. Redirects are waste of time, increases response time.</li>
	<li><b>Use of google.com URLs or use of sites that are not under *.jenkins.io for plugin should be changed to something reliable or trustworthy</b> - This is another red flag raised by SecOps and we need to remove use of google.com starting from checkConnectionURL() to anyplace it is not required. If it is really needed we may need to document it with what are the domain (full address) has to request to apply in allowed list on companies firewall on the cloud (applies to on prem too).</li>
</ol>

<p><b>Advantages</b></p>
<ol>
	<li>Increase in upgrade - Most enterprise may be running on old version of Jenkins without upgrading because these  firewall restrictions on HTTP or google.com will cause implementation failure and hard to rollback.</li>
	<li>Builds more trust in Enterprise space - It will help Jenkins to be consumed by many enterprise and management would not think of other cloud options, due to this security issues.</li>
	<li>More web traffic to Jenkins.io - More hits to Jenkins.io website if we shift from using google.com and indirectly it may help SEO and many other digital marketing advantages.</li>
</ol>
</pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [high-level-check.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/55198/high-level-check.png)
  > ![high-level-check.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/55198/high-level-check.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411455&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411455">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for raising the issue.</p>

<p>Can you describe specific cases where you're seeing that an HTTP endpoint is used?  </p>

<p>I searched the Jenkins core source code and did not find any obvious locations where it was requesting data over HTTP.  Some users make a configuration error and configure their Jenkins update center to use HTTP (as in <a href="http://updates.jenkins.io/update-center.json" class="external-link" target="_blank" rel="nofollow noopener">http://updates.jenkins.io/update-center.json</a>) when it defaults to HTTPS (as in <a href="https://updates.jenkins.io/update-center.json" class="external-link" target="_blank" rel="nofollow noopener">https://updates.jenkins.io/update-center.json</a>) and is recommended to use HTTPS in the documentation at <a href="http://updates.jenkins.io/" class="external-link" target="_blank" rel="nofollow noopener">http://updates.jenkins.io/</a> .</p>

<p>Can you describe further in what cases a plugin uses google.com?  </p>

<p>The phrase "Use of google.com URLs or use of sites that are not under *.jenkins.io for plugin" does not provide enough details to understand if your concern is with a specific plugin, with a group of plugins, or with something in Jenkins core that is acting on behalf of a plugin.</p>
</pre>
</details>

</ul>
</details>

<p>Thanks for raising the issue.</p>

<p>Can you describe specific cases where you're seeing that an HTTP endpoint is used?  </p>

<p>I searched the Jenkins core source code and did not find any obvious locations where it was requesting data over HTTP.  Some users make a configuration error and configure their Jenkins update center to use HTTP (as in <a href="http://updates.jenkins.io/update-center.json" class="external-link" target="_blank" rel="nofollow noopener">http://updates.jenkins.io/update-center.json</a>) when it defaults to HTTPS (as in <a href="https://updates.jenkins.io/update-center.json" class="external-link" target="_blank" rel="nofollow noopener">https://updates.jenkins.io/update-center.json</a>) and is recommended to use HTTPS in the documentation at <a href="http://updates.jenkins.io/" class="external-link" target="_blank" rel="nofollow noopener">http://updates.jenkins.io/</a> .</p>

<p>Can you describe further in what cases a plugin uses google.com?  </p>

<p>The phrase "Use of google.com URLs or use of sites that are not under *.jenkins.io for plugin" does not provide enough details to understand if your concern is with a specific plugin, with a group of plugins, or with something in Jenkins core that is acting on behalf of a plugin.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>prnam:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411457&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411457">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Apologies for not providing enough information, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>.</p>

<p>I am not sure of the source code difference by core etc. Please redirect if this raised on wrong component. </p>

<p>This is noticed when we try to install the Jenkins binary on any cloud with enterprise having firewalls setup such that it won't use google.com or any proxy site for plugin download or first time plugin selection download. Installer at the beginning of installation checks the internet connectivity by hitting <a href="http://www.google.com" class="external-link" target="_blank" rel="nofollow noopener">http://www.google.com</a> or <a href="http://google.com" class="external-link" target="_blank" rel="nofollow noopener">http://google.com</a> and further on plugins go to non jenkins.io domain if let's say all the plugins in custom plugins/hpi options are selected. Attached the screenshot were I can see all reference http and google.com.</p>

<p>W.r.t plugins.. it is redirecting from Jenkins.io to some other sub site to download plugins.. its random selection so did not capture all the details and stored it locally. Destroyed the VM and the project, moved with alternates on cloud solutions as it is time sensitive.</p>

<p>Troubleshoot step:
1. Allow only Jenkins.io in your firewall and rest all blocked on your VM.
2. Install Jenkins now on your VM (preferred particular cloud instance so that it will easy to see that VM specific blocks using ip address)
3. you will not be able to install successfully or you would get it installed with lot of issues or features missing which you cannot update later too without allowing temporary access to internet for selected VM or VNet.</p>
</pre>
</details>

</ul>
</details>

<p>Apologies for not providing enough information, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>.</p>

<p>I am not sure of the source code difference by core etc. Please redirect if this raised on wrong component. </p>

<p>This is noticed when we try to install the Jenkins binary on any cloud with enterprise having firewalls setup such that it won't use google.com or any proxy site for plugin download or first time plugin selection download. Installer at the beginning of installation checks the internet connectivity by hitting <a href="http://www.google.com" class="external-link" target="_blank" rel="nofollow noopener">http://www.google.com</a> or <a href="http://google.com" class="external-link" target="_blank" rel="nofollow noopener">http://google.com</a> and further on plugins go to non jenkins.io domain if let's say all the plugins in custom plugins/hpi options are selected. Attached the screenshot were I can see all reference http and google.com.</p>

<p>W.r.t plugins.. it is redirecting from Jenkins.io to some other sub site to download plugins.. its random selection so did not capture all the details and stored it locally. Destroyed the VM and the project, moved with alternates on cloud solutions as it is time sensitive.</p>

<p>Troubleshoot step:<br/>
1. Allow only Jenkins.io in your firewall and rest all blocked on your VM.<br/>
2. Install Jenkins now on your VM (preferred particular cloud instance so that it will easy to see that VM specific blocks using ip address)<br/>
3. you will not be able to install successfully or you would get it installed with lot of issues or features missing which you cannot update later too without allowing temporary access to internet for selected VM or VNet.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411462&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411462">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Based on your description, it sounds like your information security team wants more stringent access rules than most open source projects support.</p>

<p>As an example, those firewall rules will fail for access to Apache tools like Apache Maven or Apache Ant because they rely on donor mirror sites in much the same way that Jenkins relies on donor mirror sites.  The Jenkins project relies on donor mirror sites because it allows mirror sites to be placed nearer to users without requiring that the Jenkins project pay all the bandwidth costs of delivering Jenkins updates and plugin updates to over 300 000 active Jenkins installations.</p>

<p>The Jenkins project is not likely to adopt a policy of moving all mirror sites inside the jenkins.io domain.  The cost would be too great and the administrative overhead would be too great.</p>

<p>Your options:</p>

<ul>
	<li>Identify the Jenkins mirror(s) nearest to your Jenkins controller and allow access to them through your firewall definition.  You can see the mirrorlist through URL's like <a href="https://get.jenkins.io/plugins/git/4.7.2/git.hpi?mirrorlist" class="external-link" target="_blank" rel="nofollow noopener">https://get.jenkins.io/plugins/git/4.7.2/git.hpi?mirrorlist</a></li>
	<li>Create a Jenkins installation from a computer that does not have those restrictive firewall rules, then copy the plugin files to the destination computer.  Treat the computer with the restrictive firewall as though it were an offline installation</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Based on your description, it sounds like your information security team wants more stringent access rules than most open source projects support.</p>

<p>As an example, those firewall rules will fail for access to Apache tools like Apache Maven or Apache Ant because they rely on donor mirror sites in much the same way that Jenkins relies on donor mirror sites.  The Jenkins project relies on donor mirror sites because it allows mirror sites to be placed nearer to users without requiring that the Jenkins project pay all the bandwidth costs of delivering Jenkins updates and plugin updates to over 300 000 active Jenkins installations.</p>

<p>The Jenkins project is not likely to adopt a policy of moving all mirror sites inside the jenkins.io domain.  The cost would be too great and the administrative overhead would be too great.</p>

<p>Your options:</p>

<ul>
	<li>Identify the Jenkins mirror(s) nearest to your Jenkins controller and allow access to them through your firewall definition.  You can see the mirrorlist through URL's like <a href="https://get.jenkins.io/plugins/git/4.7.2/git.hpi?mirrorlist" class="external-link" target="_blank" rel="nofollow noopener">https://get.jenkins.io/plugins/git/4.7.2/git.hpi?mirrorlist</a></li>
	<li>Create a Jenkins installation from a computer that does not have those restrictive firewall rules, then copy the plugin files to the destination computer.  Treat the computer with the restrictive firewall as though it were an offline installation</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>prnam:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411467&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411467">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes, mirror sites are good and I totally understand the Jenkins point of view, but it would be great if we have option to override it to major site only, when it comes to enterprise use. This way we do not have to temporarily allow it during first install of Jenkins and remove once done .. also reduces taking lot of approvals every time we require it be allow list while we do update or upgrade. </p>

<p>Another issue with mirror site is it may have embedded trackers, those urls or redirects etc. is huge concern; we tried to enable for short period and noticed many more domains needed to be added, it was too much of task &amp; risk to keep enabling every time we see new url blocks generating while install of the Jenkins.</p>

<p>Offline installation is something that we tried but it was not something good deployment experience, having a option within install wizard to select offline installation is nice to have, IFTT. </p>

<p>Now.. above answers were focused on plugins - but the google usage.. I believe is still not addressed - today Jenkins first installation will at least hit <b>google.com</b> using HTTP for checking internet connection is active or not and then gives option to proceed next step. So I think we need to remove google.com and use <b>Jenkins.io</b> or <b>GitHub.com</b> here and of course the HTTPS. Thoughts on that?</p>
</pre>
</details>

</ul>
</details>

<p>Yes, mirror sites are good and I totally understand the Jenkins point of view, but it would be great if we have option to override it to major site only, when it comes to enterprise use. This way we do not have to temporarily allow it during first install of Jenkins and remove once done .. also reduces taking lot of approvals every time we require it be allow list while we do update or upgrade. </p>

<p>Another issue with mirror site is it may have embedded trackers, those urls or redirects etc. is huge concern; we tried to enable for short period and noticed many more domains needed to be added, it was too much of task & risk to keep enabling every time we see new url blocks generating while install of the Jenkins.</p>

<p>Offline installation is something that we tried but it was not something good deployment experience, having a option within install wizard to select offline installation is nice to have, IFTT. </p>

<p>Now.. above answers were focused on plugins - but the google usage.. I believe is still not addressed - today Jenkins first installation will at least hit <b>google.com</b> using HTTP for checking internet connection is active or not and then gives option to proceed next step. So I think we need to remove google.com and use <b>Jenkins.io</b> or <b>GitHub.com</b> here and of course the HTTPS. Thoughts on that?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411470&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411470">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>That URL is used to check for an internet connection.  I don't see the value of replacing <a href="http://www.google.com/" class="external-link" target="_blank" rel="nofollow noopener">http://www.google.com/</a> with another site that will then invoke the same concern from other users, "we don't allow access to site x".  Your information security team refuses to allow HTTP access to google.com.  Other information security teams refuse to allow HTTPS access to github.com.  Any validation URL we select has a risk that some information security team will forbid access to that URL.  It might be better to use <a href="https://www.google.com" class="external-link" target="_blank" rel="nofollow noopener">https://www.google.com</a> than <a href="http://www.google.com" class="external-link" target="_blank" rel="nofollow noopener">http://www.google.com</a> since the update center URL's are all using HTTPS now, but that change won't be enough to solve your issue.</p>
</pre>
</details>

</ul>
</details>

<p>That URL is used to check for an internet connection.  I don't see the value of replacing <a href="http://www.google.com/" class="external-link" target="_blank" rel="nofollow noopener">http://www.google.com/</a> with another site that will then invoke the same concern from other users, "we don't allow access to site x".  Your information security team refuses to allow HTTP access to google.com.  Other information security teams refuse to allow HTTPS access to github.com.  Any validation URL we select has a risk that some information security team will forbid access to that URL.  It might be better to use <a href="https://www.google.com" class="external-link" target="_blank" rel="nofollow noopener">https://www.google.com</a> than <a href="http://www.google.com" class="external-link" target="_blank" rel="nofollow noopener">http://www.google.com</a> since the update center URL's are all using HTTPS now, but that change won't be enough to solve your issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>prnam:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411471&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411471">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am confident that any information security team won't refuse <a href="https://www.jenkins.io" class="external-link" target="_blank" rel="nofollow noopener">https://www.jenkins.io </a>site  <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>I am confident that any information security team won't refuse <a href="https://www.jenkins.io" class="external-link" target="_blank" rel="nofollow noopener">https://www.jenkins.io </a>site  <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>prnam:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411472&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411472">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>If team is going to change value in update center then, I feel its better we use <a href="https://www.jenkins.io" class="external-link" target="_blank" rel="nofollow noopener">https://www.jenkins.io</a> than <a href="https://www.google.com" class="external-link" target="_blank" rel="nofollow noopener">https://www.google.com</a></p>
</pre>
</details>

</ul>
</details>

<p>If team is going to change value in update center then, I feel its better we use <a href="https://www.jenkins.io" class="external-link" target="_blank" rel="nofollow noopener">https://www.jenkins.io</a> than <a href="https://www.google.com" class="external-link" target="_blank" rel="nofollow noopener">https://www.google.com</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/timja.png" title="timja's avatar" /> timja:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411481&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411481">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It's likely better that we use a domain under our control that must be used as part of the download process already.</p>

<p>Opening google on the firewall will likely frustrate people who have no need for their Jenkins to reach google.</p>

<p>I wouldn't default to recommending www.jenkins.io as it's not needed during download.</p>

<p>updates.jenkins.io is likely best as it doesn't seem to have HSTS enabled.</p>

<p>Related issue here: <a href="https://github.com/jenkinsci/docker/issues/973#issuecomment-668955696" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/docker/issues/973#issuecomment-668955696</a></p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> any thoughts?</p>


<p>&#8212;</p>

<p>RE the mirror sites, they are 'documented' at <a href="https://github.com/jenkins-infra/charts/blob/master/charts/mirrorbits/README.md#add-mirrors-configuration" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkins-infra/charts/blob/master/charts/mirrorbits/README.md#add-mirrors-configuration</a> you can whitelist all of them or just the ones closest to you and hope that you don't get redirected to ones that don't work for you</p>


</pre>
</details>

</ul>
</details>

<p>It's likely better that we use a domain under our control that must be used as part of the download process already.</p>

<p>Opening google on the firewall will likely frustrate people who have no need for their Jenkins to reach google.</p>

<p>I wouldn't default to recommending www.jenkins.io as it's not needed during download.</p>

<p>updates.jenkins.io is likely best as it doesn't seem to have HSTS enabled.</p>

<p>Related issue here: <a href="https://github.com/jenkinsci/docker/issues/973#issuecomment-668955696" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/docker/issues/973#issuecomment-668955696</a></p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> any thoughts?</p>


<p>&#8212;</p>

<p>RE the mirror sites, they are 'documented' at <a href="https://github.com/jenkins-infra/charts/blob/master/charts/mirrorbits/README.md#add-mirrors-configuration" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkins-infra/charts/blob/master/charts/mirrorbits/README.md#add-mirrors-configuration</a> you can whitelist all of them or just the ones closest to you and hope that you don't get redirected to ones that don't work for you</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411482&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411482">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>It's likely better that we use a domain under our control that must be used as part of the download process already.</p></blockquote>
<p>The point of the test, AFAIUI, is to be able to distinguish between "Jenkins infra is inaccessible" and "internet is inaccessible", and using the same domain we're downloading things from anyway wouldn't do that.</p>

<p> </p>
<blockquote><p>I don't see the value of replacing <a href="http://www.google.com/" class="external-link" target="_blank" rel="nofollow noopener">http://www.google.com/</a> with another site that will then invoke the same concern from other users</p></blockquote>
<p>A while back I looked at moving to <a href="http://www.msftncsi.com/ncsi.txt" class="external-link" target="_blank" rel="nofollow noopener">http://www.msftncsi.com/ncsi.txt</a> (which is a domain and URL dedicated to Windows's network connection status indicator), but it required more rework than I was willing to do at the time. That seems like a mostly reasonable compromise to me.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>It's likely better that we use a domain under our control that must be used as part of the download process already.</p></blockquote>
<p>The point of the test, AFAIUI, is to be able to distinguish between "Jenkins infra is inaccessible" and "internet is inaccessible", and using the same domain we're downloading things from anyway wouldn't do that.</p>

<p> </p>
<blockquote><p>I don't see the value of replacing <a href="http://www.google.com/" class="external-link" target="_blank" rel="nofollow noopener">http://www.google.com/</a> with another site that will then invoke the same concern from other users</p></blockquote>
<p>A while back I looked at moving to <a href="http://www.msftncsi.com/ncsi.txt" class="external-link" target="_blank" rel="nofollow noopener">http://www.msftncsi.com/ncsi.txt</a> (which is a domain and URL dedicated to Windows's network connection status indicator), but it required more rework than I was willing to do at the time. That seems like a mostly reasonable compromise to me.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kon:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411493&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411493">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is the Internet connectivity check (using Google) only intended to help an admin diagnose why Jenkins is not able to contact the update server? If so, perhaps Jenkins could skip this check by default but let the admin click a button to deliberately trigger the check. Or is it needed for telemetry as well?</p>
</pre>
</details>

</ul>
</details>

<p>Is the Internet connectivity check (using Google) only intended to help an admin diagnose why Jenkins is not able to contact the update server? If so, perhaps Jenkins could skip this check by default but let the admin click a button to deliberately trigger the check. Or is it needed for telemetry as well?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411496&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411496">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>only intended to help an admin diagnose why Jenkins is not able to contact the update server?</p></blockquote>
<p>Yes.</p>
<blockquote><p>If so, perhaps Jenkins could skip this check by default but let the admin click a button to deliberately trigger the check.</p></blockquote>
<p>That might work (although the design of the update center is really weird so might be more work than one would expect, unsure). Ideally this would be implemented in such a (visible) way that we don't end up getting tons of reports about broken infra when really the networking / proxy settings / etc. are at fault.</p>
<blockquote><p>Or is it needed for telemetry as well?</p></blockquote>
<p>No, we don't collect that. Anonymous usage stats exist but they submit data, or don't (and what we collect is fairly minimal to begin with).</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>only intended to help an admin diagnose why Jenkins is not able to contact the update server?</p></blockquote>
<p>Yes.</p>
<blockquote><p>If so, perhaps Jenkins could skip this check by default but let the admin click a button to deliberately trigger the check.</p></blockquote>
<p>That might work (although the design of the update center is really weird so might be more work than one would expect, unsure). Ideally this would be implemented in such a (visible) way that we don't end up getting tons of reports about broken infra when really the networking / proxy settings / etc. are at fault.</p>
<blockquote><p>Or is it needed for telemetry as well?</p></blockquote>
<p>No, we don't collect that. Anonymous usage stats exist but they submit data, or don't (and what we collect is fairly minimal to begin with).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>prnam:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411500&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411500">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>A while back I looked at moving to <a href="http://www.msftncsi.com/ncsi.txt" class="external-link" target="_blank" rel="nofollow noopener">http://www.msftncsi.com/ncsi.txt</a> (which is a domain and URL dedicated to Windows's network connection status indicator), but it required more rework than I was willing to do at the time. That seems like a mostly reasonable compromise to me.</p></blockquote>
<blockquote><p>only intended to help an admin diagnose why Jenkins is not able to contact the update server?</p>
<ul class="alternate" type="square">
	<li>Yes.</li>
</ul>
</blockquote>
<blockquote><p>The point of the test, AFAIUI, is to be able to distinguish between "Jenkins infra is inaccessible" and "internet is inaccessible", and using the same domain we're downloading things from anyway wouldn't do that.</p></blockquote>
<p>I would vote for URL that is close to open source community, something like Jenkins project repo link in GitHub.com or The Linux Foundation site URL that speaks about Jenkins or CD Foundation with no cookies or trackers or cache sent with request, regardless what's decide between manual check for admin using button or as-is but just URL update.</p>

<p>These URL requests are more trustworthy than google or msft or any other commercial URLs (includes DDG) with trackers and cookies that stay forever if admins have no idea something of this happening at the behind of Jenkins.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>A while back I looked at moving to <a href="http://www.msftncsi.com/ncsi.txt" class="external-link" target="_blank" rel="nofollow noopener">http://www.msftncsi.com/ncsi.txt</a> (which is a domain and URL dedicated to Windows's network connection status indicator), but it required more rework than I was willing to do at the time. That seems like a mostly reasonable compromise to me.</p></blockquote>
<blockquote><p>only intended to help an admin diagnose why Jenkins is not able to contact the update server?</p>
<ul class="alternate" type="square">
	<li>Yes.</li>
</ul>
</blockquote>
<blockquote><p>The point of the test, AFAIUI, is to be able to distinguish between "Jenkins infra is inaccessible" and "internet is inaccessible", and using the same domain we're downloading things from anyway wouldn't do that.</p></blockquote>
<p>I would vote for URL that is close to open source community, something like Jenkins project repo link in GitHub.com or The Linux Foundation site URL that speaks about Jenkins or CD Foundation with no cookies or trackers or cache sent with request, regardless what's decide between manual check for admin using button or as-is but just URL update.</p>

<p>These URL requests are more trustworthy than google or msft or any other commercial URLs (includes DDG) with trackers and cookies that stay forever if admins have no idea something of this happening at the behind of Jenkins.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kon:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411503&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411503">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is this the connection check we’re discussing? <a href="https://github.com/jenkinsci/jenkins/blob/bd7c823c2781d9282eab71a5e2967d386c629344/core/src/main/java/hudson/model/UpdateCenter.java#L1619" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/blob/bd7c823c2781d9282eab71a5e2967d386c629344/core/src/main/java/hudson/model/UpdateCenter.java#L1619</a> indirectly calling <a href="https://github.com/jenkinsci/jenkins/blob/bd7c823c2781d9282eab71a5e2967d386c629344/core/src/main/java/hudson/model/UpdateCenter.java#L1403" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/blob/bd7c823c2781d9282eab71a5e2967d386c629344/core/src/main/java/hudson/model/UpdateCenter.java#L1403</a></p>

<p>If Google sets some cookies in the HTTP response, then I don’t think Jenkins saves them anywhere at all. It does not run any scripts or load any referenced images or style sheets, either.</p>
</pre>
</details>

</ul>
</details>

<p>Is this the connection check we’re discussing? <a href="https://github.com/jenkinsci/jenkins/blob/bd7c823c2781d9282eab71a5e2967d386c629344/core/src/main/java/hudson/model/UpdateCenter.java#L1619" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/blob/bd7c823c2781d9282eab71a5e2967d386c629344/core/src/main/java/hudson/model/UpdateCenter.java#L1619</a> indirectly calling <a href="https://github.com/jenkinsci/jenkins/blob/bd7c823c2781d9282eab71a5e2967d386c629344/core/src/main/java/hudson/model/UpdateCenter.java#L1403" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/blob/bd7c823c2781d9282eab71a5e2967d386c629344/core/src/main/java/hudson/model/UpdateCenter.java#L1403</a></p>

<p>If Google sets some cookies in the HTTP response, then I don’t think Jenkins saves them anywhere at all. It does not run any scripts or load any referenced images or style sheets, either.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411506&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411506">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>close to open source community, something like Jenkins project repo link in GitHub.com</p></blockquote>
<p>I don't think declining to use a domain used by Microsoft exclusively (AFAICT) for their operating system connection test, and instead using the URL of one of their marketing and analytics contaminated products is a good idea, unless you have specific reasons that msftncsi would be a bad choice.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>close to open source community, something like Jenkins project repo link in GitHub.com</p></blockquote>
<p>I don't think declining to use a domain used by Microsoft exclusively (AFAICT) for their operating system connection test, and instead using the URL of one of their marketing and analytics contaminated products is a good idea, unless you have specific reasons that msftncsi would be a bad choice.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kon:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411507&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411507">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am a bit concerned that a captive portal system might special-case the Microsoft connection test URLs somehow and thereby interfere with the test. I cannot think of a good reason to do so, though.</p>
</pre>
</details>

</ul>
</details>

<p>I am a bit concerned that a captive portal system might special-case the Microsoft connection test URLs somehow and thereby interfere with the test. I cannot think of a good reason to do so, though.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411508&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411508">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>If Google sets some cookies in the HTTP response, then I don’t think Jenkins saves them anywhere at all. It does not run any scripts or load any referenced images or style sheets, either.</p></blockquote>
<p>Correct, we send a single request and expect a success response of which we only check the status code. No cookies etc are stored etc., this isn't happening in a browser.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>If Google sets some cookies in the HTTP response, then I don’t think Jenkins saves them anywhere at all. It does not run any scripts or load any referenced images or style sheets, either.</p></blockquote>
<p>Correct, we send a single request and expect a success response of which we only check the status code. No cookies etc are stored etc., this isn't happening in a browser.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411509&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411509">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Getting back to the previous comment by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=timja" class="user-hover" rel="timja">timja</a>, there is one more oddity in this mess:</p>

<p>Historically, Jenkins didn't download the update center JSON itself, a user's browser did and told Jenkins about it. The reason is that apparently, in the late 2000s, it wasn't common for services like Jenkins to connect out, so a bit of client-side JS did it for Jenkins. Downloading plugins was done from Jenkins though (so a restricted environment may know  there are plugin updates, but could not install them).</p>

<p>In this environment, a separate connection test made sense. Perhaps that's kind of obsolete now, and we should have an admin monitor informing users their updated-daily update site metadata could not be downloaded instead? Then the connection test could just be removed, given that lack of internet access means there's no metadata. I think I have some work in progress locally somewhere, if I find it I'll open a PR for that.</p>
</pre>
</details>

</ul>
</details>

<p>Getting back to the previous comment by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=timja" class="user-hover" rel="timja">timja</a>, there is one more oddity in this mess:</p>

<p>Historically, Jenkins didn't download the update center JSON itself, a user's browser did and told Jenkins about it. The reason is that apparently, in the late 2000s, it wasn't common for services like Jenkins to connect out, so a bit of client-side JS did it for Jenkins. Downloading plugins was done from Jenkins though (so a restricted environment may know  there are plugin updates, but could not install them).</p>

<p>In this environment, a separate connection test made sense. Perhaps that's kind of obsolete now, and we should have an admin monitor informing users their updated-daily update site metadata could not be downloaded instead? Then the connection test could just be removed, given that lack of internet access means there's no metadata. I think I have some work in progress locally somewhere, if I find it I'll open a PR for that.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>prnam:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411873&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411873">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Also, if you think from receiving end (google) it looks like they are having DDoS attack, not sure of the client install but if one is using via docker without persistence this could be bad for person who was experimenting stuff or developing and testing.</p>

<p>OR it also can be that Jenkins in different way providing this signal to google of all new users connect at least once org or personal.. and contribute for their daily or yearly active users.</p>

<p>I think there should be better mechanism for achieving what it was built for and if it’s obsolete then it will be nice if team includes fix in next release or have it on roadmap. I wish I could contribute but unfortunately I am not java developer.</p>
</pre>
</details>

</ul>
</details>

<p>Also, if you think from receiving end (google) it looks like they are having DDoS attack, not sure of the client install but if one is using via docker without persistence this could be bad for person who was experimenting stuff or developing and testing.</p>

<p>OR it also can be that Jenkins in different way providing this signal to google of all new users connect at least once org or personal.. and contribute for their daily or yearly active users.</p>

<p>I think there should be better mechanism for achieving what it was built for and if it’s obsolete then it will be nice if team includes fix in next release or have it on roadmap. I wish I could contribute but unfortunately I am not java developer.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>prnam:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411874&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411874">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>Correct, we send a single request and expect a success response of which we only check the status code. No cookies etc are stored etc., this isn't happening in a browser.</p></blockquote>

<p>We are assuming best case here.. even if Jenkins is not hitting service on http on google or storing them; there are so many other cookies and trackers on machine of some individual may be giving way to do so many different commercial benefits; may be the same commercial company that does what Jenkins does will be notified about the traffic and interest of this particular individual or company on using this Jenkins may be interested in using your product and persuade with different marketing strategy to convert current jenkins user to possible customer for their platforms.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>Correct, we send a single request and expect a success response of which we only check the status code. No cookies etc are stored etc., this isn't happening in a browser.</p></blockquote>

<p>We are assuming best case here.. even if Jenkins is not hitting service on http on google or storing them; there are so many other cookies and trackers on machine of some individual may be giving way to do so many different commercial benefits; may be the same commercial company that does what Jenkins does will be notified about the traffic and interest of this particular individual or company on using this Jenkins may be interested in using your product and persuade with different marketing strategy to convert current jenkins user to possible customer for their platforms.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kon:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411875&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411875">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=prnam" class="user-hover" rel="prnam">prnam</a> can you get a network packet capture and check whether the HTTP request indicates in any way that it is sent by Jenkins?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=prnam" class="user-hover" rel="prnam">prnam</a> can you get a network packet capture and check whether the HTTP request indicates in any way that it is sent by Jenkins?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66161?focusedId=411919&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-411919">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>cookies and trackers on machine</p></blockquote>
<p>That's not how any of this works. They get an IP address. Depending on Jenkins version and configuration they may also get a "Jenkins" user agent. But no "cookies and trackers".</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>cookies and trackers on machine</p></blockquote>
<p>That's not how any of this works. They get an IP address. Depending on Jenkins version and configuration they may also get a "Jenkins" user agent. But no "cookies and trackers".</p>
